processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -254,6 +254,38 @@ public File merge(
     return merge(indexes, metricAggs, outDir, segmentMetadata, indexSpec, new BaseProgressIndicator());
   }
 
+  private List<String> getLexicographicMergedDimensions(List<IndexableAdapter> indexes)
+  {
+    return mergeIndexed(
+        Lists.transform(
+            indexes,
+            new Function<IndexableAdapter, Iterable<String>>()
+            {
+              @Override
+              public Iterable<String> apply(@Nullable IndexableAdapter input)
+              {
+                return input.getDimensionNames();
+              }
+            }
+        )
+    );
+  }
+
+  private List<String> getMergedDimensions(List<IndexableAdapter> indexes)
+  {
+    if (indexes.size() == 0) {
+      return new ArrayList<>();
+    }
+    List<String> dimOrder = Lists.newArrayList(indexes.get(0).getDimensionNames().iterator());
```

Comment:
You don't need the `.iterator()`. Also, you could use `ImmutableList.copyOf` instead, since immutable lists are cool.